web2.0/src/Components/Historic/HistData.js

```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect } from 'react';
+import WeatherCard from '../weather-card/WeatherCard';
+
+const NUM_DATA_POINTS = 12;
+
+const HistData = () => {
+  const [currentData, setCurrentData] = useState();
+
+  const getData = async () => {
+    const windResponse = await fetch('http://localhost:8000/wind');
+    const windData = await windResponse.json();
+    const swellResponse = await fetch('http://localhost:8000/swell');
+    const swellData = await swellResponse.json();
+    setCurrentData({ windData, swellData });
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const makeHistData = () => {
+    const histData = [];
+
+    for (let i = 0; i < NUM_DATA_POINTS; i++) {
```

Comment:
Mind helping me understand the purpose for `NUM_DATA_POINTS` in your loop? Are you restricting the number of records (i.e., rows) to the most recent 12?

Reply:
Oversight, should've just been currentData.length instead.

Reply:
No worries. It happens to me all the time too!

Reply:
> ... should've just been currentData.length instead.

@Grieze - I try to build the habit of always questioning what is type (i.e., what data structure) of object I'm working with at any given time. So, in this instance, I would ask myself:
- Does `currentData` represent a single row or many rows? The answer will influence how I set `.length` in order to control the loop.

Side note: ES6+ introduces different ways to iterate over objects:
- [.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)
- [.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

You might have an easier time refactoring your code with that approach. Let me know what you think. What are the pros/cons of using the above compared to how you're iterating?